request-manager/graph/grapher.go

```diff
@@ -272,13 +294,14 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 
 				delete(nodesToAdd, nodeName)
 				nodesAdded[nodeName] = true
-				componentAdded = true
+				nodeAdded = true
 			}
 
 		}
 
-		// If we were unable to add nodes, there must be a cyclical dependency, which is an error
-		if !componentAdded {
+		// If we were unable to add nodes, there must be a cyclical
+		// dependency, which is an error
+		if !nodeAdded {
```

Comment:
Is this correct? For example,
```
A->B->C -+
       ^-----/
```
Cycle between B and C but if we build A first then `nodeAdded` will be true. Perhaps this needs to be `if len(nodesToAdd) != 0`?

Reply:
Re previous comments and Slack message: I see now. So we can "solve" this with some illuminating code comments (either here are maybe on the inner loop), something to the effect of:
```go
// Each loop through the nodes, we should be able to add at least 1 node.
// If not, the there's a cycle (or perhaps a typo in deps: that wasn't caught
// earlier). For example, with A->B->C, 1st loop will add A since it has no
// deps, then it'll add B since it depends on A which has been built, then C.
// But if a cycle existed between B and C, the 2nd loop wouldn't be able to
// build B and nodeAdded would be false and trigger the error after the loop.
```